tutorials/Pattern Recognition.ipynb

```diff
@@ -0,0 +1,739 @@
+    "!pip install pandas"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "import datetime\n",
+    "from pyclarify.client import APIClient\n",
+    "from pyclarify.models.data import DataFrame\n",
+    "from pyclarify.models.data import Signal\n",
+    "from pyclarify.models.requests import (\n",
+    "        ItemSelect,\n",
+    "        SelectItemsParams,\n",
+    "        SelectTimesParams,\n",
+    "        SelectSeriesParams,\n",
+    ")\n",
+    "import plotly.graph_objects as go\n",
+    "import plotly.express as px\n",
+    "import math"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "client = APIClient(\"./clarify-credentials.json\")\n",
+    "\n",
+    "item_id = \"<item-id>\"\n",
+    "start_date = \"<stating date>\"   # example: \"2021-04-17T03:22:06Z\"\n",
+    "end_date = \"<ending date>\"      # example: \"2021-08-12T22:00:06Z\"\n",
+    "\n",
+    "items = SelectItemsParams(include=True, filter={\"id\": {\"$in\": [item_id]}})\n",
+    "series = SelectSeriesParams(items=True)\n",
+    "\n",
+    "# The API returns data with a window of 40 days.\n",
+    "start = datetime.datetime.strptime(start_date,\"%Y-%m-%dT%H:%M:%SZ\")\n",
+    "end = datetime.datetime.strptime(end_date,\"%Y-%m-%dT%H:%M:%SZ\")\n",
+    "dt = end - start\n",
+    "\n",
+    "hours = dt.days * 24  + dt.seconds//3600 + 1\n",
+    "blocks = math.ceil(hours / 960)\n",
+    "delta = (end - start)/blocks\n",
+    "\n",
+    "date_list = [start.strftime(\"%Y-%m-%dT%H:%M:%S\")+'Z']\n",
+    "\n",
+    "for block in range(1, blocks + 1):\n",
+    "    time = (start + block * delta).strftime(\"%Y-%m-%dT%H:%M:%S\")+'Z'\n",
+    "    date_list.append(time)\n",
+    "\n",
+    "values = []\n",
+    "dates = []\n",
+    "\n",
+    "for date in range(len(date_list)-1):\n",
+    "\n",
+    "    times = SelectTimesParams(notBefore=date_list[date], before=date_list[date+1])\n",
+    "    params = ItemSelect(items=items, times=times, series=series)\n",
+    "    response = client.select_items(params=params)\n",
+    "    \n",
+    "    values.extend(list(response.result.data.series.values())[0])\n",
+    "    dates.extend(response.result.data.times)"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Plot your data. \n",
+    "You shoud see a similar plot as you see it Clarify. Keep in mind that the plot shape slidely changes as you zoom in and out in Clarify."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "fig = go.Figure([go.Scatter(x=dates, y=values )])\n",
+    "fig.show()"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "__Time to mark the pattern of interest (or else query) in the plot.__"
```

Comment:
Suggest rewriting. E.g Time to mark the _pattern of interest_, sometimes called _query_, in the plot

Reply:
yea 👍